packages/studio/src/components/ComponentTree.tsx

```diff
@@ -167,32 +204,118 @@ function useTree(): NodeModel<ComponentState>[] | undefined {
 }
 
 function useDropHandler() {
-  const updateComponentTree = useStudioStore(
-    (store) => store.actions.updateComponentTree
-  );
+  const [selectedComponentUUIDs, componentTree, updateComponentTree] =
+    useStudioStore((store) => {
+      return [
+        store.pages.selectedComponentUUIDs,
+        store.actions.getComponentTree(),
+        store.actions.updateComponentTree,
+      ];
+    });
 
-  const handleDrop = useCallback(
-    (tree: NodeModel<ComponentState>[]) => {
-      const updatedComponentTree = tree.map((n) => {
-        if (!n.data) {
-          throw new Error(
-            "Unable to handle drag and drop event in ComponentTree: " +
-              "No data passed into NodeModel<ComponentState> for node " +
-              JSON.stringify(n, null, 2)
-          );
-        }
-        const componentState: ComponentState = {
-          ...n.data,
-          parentUUID: n.parent.toString(),
-        };
-        if (componentState.parentUUID === ROOT_ID) {
-          delete componentState.parentUUID;
+  const handleSelectedDrop = useCallback(
+    (
+      updatedComponentTree: ComponentState[],
+      dragSourceId: string,
+      destinationParentId: string
+    ) => {
+      if (!componentTree) {
+        throw new Error(
+          "Unable to handle drag and drop event: component tree is undefined."
+        );
+      }
+      const hasNoChanges = isEqual(
```

Comment:
for my own understanding, what's an example where `handleSelectedDrop` would be called but `hasNoChanges` is true? I see that in `handleDrop` we already check that the destination is not one of the selected components before calling `handleSelectedDrop`

Reply:
I can't find any specific ones at the moment, but I remember encountering a few edge cases when dragging components around that were only handled by checking `hasNoChanges`. Only checking that the destination is not a selected component is not sufficient, since the "destination" is really the parent in which the components are dropped, not the exact position. I also like the readability `hasNoChanges` and think it makes more sense than checking specific cases.

Reply:
I don't think we can remove the destination check because the tree could still have changes if you drag one of the selected components above another selected component (I'm seeing [this](https://github.com/yext/studio/pull/303#issuecomment-1672018548) behavior). But in that case, we wouldn't want to update anything. Can we add a test for this case?